platform/android/java/lib/src/org/godotengine/godot/input/GodotInputHandler.java

```diff
@@ -82,19 +91,20 @@ public boolean onKeyUp(final int keyCode, KeyEvent event) {
 
 		if (keyCode == KeyEvent.KEYCODE_VOLUME_UP || keyCode == KeyEvent.KEYCODE_VOLUME_DOWN) {
 			return false;
-		};
+		}
 
 		int source = event.getSource();
 		if (isKeyEvent_GameDevice(source)) {
-			final int button = getGodotButton(keyCode);
-			final int device_id = findJoystickDevice(event.getDeviceId());
-
 			// Check if the device exists
-			if (device_id > -1) {
+			final int deviceId = event.getDeviceId();
+			if (mJoystickIds.indexOfKey(deviceId) >= 0) {
```

Comment:
This is optional since they're equivalent but you could also use `mJoysticksIds.contains(deviceId)` instead.
See https://developer.android.com/reference/android/util/SparseArray#contains(int)

Reply:
I'll keep that in mind for adding to the next PR

Reply:
Added in API level 30